### PR TITLE
Sunset *_pattern node pattern helpers

### PR DESCRIPTION
**Pull request by pirj**
Follow-up to #1578
Fixes #1578

They introduce another level of indirection that is arguably harder to read.
```
send_pattern('{#Includes.examples #Examples.all}')
# vs
'(send nil? {#Includes.examples #Examples.all} ...)'

block_pattern('#ExampleGroups.all')
# vs
'(block (send #rspec? #ExampleGroups.all ...) ...)'

'(block $#{send_pattern('#Hooks.all')} _ nil?)'
# vs
'(block $(send nil? #Hooks.all ...) _ nil?)'

block_or_numblock_pattern('{#Examples.regular #Examples.focused}')
# vs
'({block numblock} (send nil? {#Examples.regular #Examples.focused} ...) ...)'
```

Those helpers implied that `#rspec?`, e.g. an explicit `RSpec` receiver or an implicit receiver can be matched, while in some cases like `let`/`before` using the explicit receiver is not supported by RSpec.

Some discussion: https://github.com/rubocop/rubocop-rspec/pull/1576#issuecomment-1421283557

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [x] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/rspec/pending_spec.rb [hunk 1]
```diff
@@ -150,14 +150,6 @@
     RUBY
   end
 
-  it 'flags pending examples when receiver is explicit' do
-    expect_offense(<<-RUBY)
-      RSpec.xit 'test' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/pending_spec.rb:155]: `RSpec.xit` won't work.

**Comment by pirj** [comment]
cc @ydah

**Review by bquorning — APPROVED** [review]
Fine by me.

@Darhazer do you have an opinion on this change?

**Review by ydah — APPROVED** [review]
Looks very good to me😀

**Review by Darhazer — APPROVED** [review]
Fine by me.
Should we do something more about the deprecation of the API (outputting a message, mentioning it in the changelog?)

**Comment by pirj** [comment]
Added deprecation warnings:
```
$ rspec $(fd described_class_sp)
/Users/pirj/source/rubocop-rspec/lib/rubocop/cop/rspec/described_class.rb:72: warning: Usage of block_pattern is deprecated. Use node pattern explicitly.
/Users/pirj/source/rubocop-rspec/lib/rubocop/rspec/language/node_pattern.rb:20: warning: Usage of send_pattern is deprecated. Use node pattern explicitly.

Randomized with seed 7673

RuboCop::Cop::RSpec::DescribedClass
...

Finished in 0.3949 seconds (files took 7.24 seconds to load)
31 examples, 0 failures

Randomized with seed 7673
````

**Pull request merged**